pkgs/development/tools/misc/ccache/default.nix

```diff
@@ -1,29 +1,23 @@
-{ stdenv, fetchurl, perl, zlib, makeWrapper }:
+{ stdenv, fetchFromGitHub, asciidoc-full, gperf, perl, autoconf, zlib, makeWrapper }:
 
 let ccache = stdenv.mkDerivation rec {
   pname = "ccache";
-  version = "3.4.1";
+  version = "3.7.10";
 
-  src = fetchurl {
-    sha256 = "1pppi4jbkkj641cdynmc35jaj40jjicw7gj75ran5qs5886jcblc";
-    url = "mirror://samba/ccache/${pname}-${version}.tar.xz";
+  src = fetchFromGitHub {
+    owner = "ccache";
+    repo = "ccache";
+    rev = "v${version}";
+    sha256 = "0v6pvj50y18fxh2f6cx6xyg545xcpxfmrcg9bzbc6lc5acrvvjm7";
   };
 
-  nativeBuildInputs = [ perl ];
+  nativeBuildInputs = [ asciidoc-full autoconf gperf perl ];
 
   buildInputs = [ zlib ];
 
   outputs = [ "out" "man" ];
 
-  # non to be fail on filesystems with unconventional blocksizes (zfs on Hydra?)
-  patches = [
-    ./fix-debug-prefix-map-suite.patch
-    ./skip-fs-dependent-test.patch
-  ];
-
-  postPatch = ''
-    substituteInPlace Makefile.in --replace 'objs) $(extra_libs)' 'objs)'
-  '';
+  preConfigure = "./autogen.sh";
```

Comment:
With `autoreconfHook` in `nativeBuildInputs`, this can be removed.

Reply:
Thanks for letting me know, looks better now.